### PR TITLE
Task-40046 : Notify author and admin when dlp items are detected

### PR DESCRIPTION
**Pull request by AzmiTouil**

### CODE REVIEW

File: extension/notification/src/main/resources/locale/notification/template/Notification_en.properties [hunk 1]
```diff
@@ -247,3 +247,10 @@ Notification.digest.more.PostActivitySpaceStreamPlugin=$LAST3_USERS <span style=
 #############################################################################
 LinkActivity.notification.label.source=Source
 LinkActivity.notification.label.WatchTheVideo=Watch the video
+        
+#############################################################################
+#                         DlpDetectedItemPlugin                             #
+#############################################################################
+
+Notification.intranet.message.DlpUserDetectedItemPlugin=The item named <strong>{0}</strong> needs to be analyzed. So this item is in quarantine at the moment
```
Comment (rdenarie) [inline extension/notification/src/main/resources/locale/notification/template/Notification_en.properties:263]: Can you create the french file also ?

File: extension/notification/src/main/webapp/WEB-INF/conf/social-notification-extension/social/notification-plugins-configuration.xml [hunk 2]
```diff
@@ -697,4 +697,89 @@
     </component-plugin>
   </external-component-plugins>
 
+  <external-component-plugins>
```
Comment (rdenarie) [inline extension/notification/src/main/webapp/WEB-INF/conf/social-notification-extension/social/notification-plugins-configuration.xml:700]: Can you check it the configuration is ok for mobile and onsite notification ?

File: extension/notification/src/main/webapp/WEB-INF/conf/social-notification-extension/social/notification-plugins-configuration.xml [hunk 3]
```diff
@@ -697,4 +697,89 @@
...
+            </field>
+            <field name="order">
+              <string>12</string>
+            </field>
+            <field name="defaultConfig">
+              <collection type="java.util.ArrayList">
+                <value>
+                  <string>Instantly</string>
+                </value>
+                <value>
```
Comment (rdenarie) [inline extension/notification/src/main/webapp/WEB-INF/conf/social-notification-extension/social/notification-plugins-configuration.xml]: not sure that daily is needed

File: extension/notification/src/main/webapp/WEB-INF/conf/social-notification-extension/social/notification-plugins-configuration.xml [hunk 4]
```diff
@@ -697,4 +697,89 @@
...
+            </field>
+            <field name="order">
+              <string>11</string>
+            </field>
+            <field name="defaultConfig">
+              <collection type="java.util.ArrayList">
+                <value>
+                  <string>Instantly</string>
+                </value>
+                <value>
```
Comment (rdenarie) [inline extension/notification/src/main/webapp/WEB-INF/conf/social-notification-extension/social/notification-plugins-configuration.xml]: not sure that daily is needed

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: extension/notification/src/main/resources/locale/notification/template/Notification_fr.properties [hunk 5]
```diff
@@ -247,3 +247,10 @@ Notification.digest.more.PostActivitySpaceStreamPlugin=$LAST3_USERS <span style=
 #############################################################################
 LinkActivity.notification.label.source=Source
 LinkActivity.notification.label.WatchTheVideo=Voir la vid\u00E9o
+
+#############################################################################
+#                         DlpDetectedItemPlugin                             #
+#############################################################################
+
+Notification.intranet.message.DlpUserDetectedItemPlugin=L'\u00E9l\u00E9ment nomm\u00E9 <strong>{0}</strong> doit \u00EAtre analys\u00E9. Donc, cet \u00E9l\u00E9ment est en quarantaine pour le moment
```
Comment (rdenarie) [inline extension/notification/src/main/resources/locale/notification/template/Notification_fr.properties]:
```suggestion
Notification.intranet.message.DlpUserDetectedItemPlugin=L'\u00E9l\u00E9ment nomm\u00E9 <strong>{0}</strong> doit \u00EAtre analys\u00E9, il a \u00E9t\u00E9 mis en quarantaine pour le moment.
```

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: component/notification/src/main/java/org/exoplatform/social/notification/LinkProviderUtils.java [hunk 6]
```diff
@@ -306,4 +306,14 @@ public static String getOpenLink(ExoSocialActivity activity) {
...
   }
+
+  /**
+   * Get the Redirect quarantine page url
+   *
+   * @return url http://localhost:8080/portal/login?initialURI=/portal/g/:platform:administrators/dlp-quarantine
+   */
+  public static String getQuarantineRedirectURL() {
+    String portal = PortalContainer.getCurrentPortalContainerName();
+    return new StringBuffer(CommonsUtils.getCurrentDomain()).append("/").append(portal).append(LOGIN_INITIALURI).append(portal).append("/").append("g/:platform:administrators/dlp-quarantine").toString();
```
Comment (rdenarie) [inline component/notification/src/main/java/org/exoplatform/social/notification/LinkProviderUtils.java]: Can you explain what is it for ?

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: component/notification/src/main/java/org/exoplatform/social/notification/LinkProviderUtils.java (hunk 6 above)
Comment (AzmiTouil) [reply component/notification/src/main/java/org/exoplatform/social/notification/LinkProviderUtils.java]: This is for redirect admin to the quarantine page from the mail template

Comment (rdenarie) [reply component/notification/src/main/java/org/exoplatform/social/notification/LinkProviderUtils.java]: Why we need to redirect him on login url ? 
What is the behaviour if you put the url of the quarantine page without login ?

Comment (AzmiTouil) [reply component/notification/src/main/java/org/exoplatform/social/notification/LinkProviderUtils.java]: The same behaviour !
Updated.

**Review by rdenarie — DISMISSED** [review]

**Review by rdenarie — APPROVED** [review]

**Pull request merged**